core/trino-main/src/test/java/io/trino/server/security/TestResourceSecurity.java

```diff
@@ -816,6 +816,48 @@ public void testJwtAndOAuth2AuthenticatorsSeparation()
         }
     }
 
+    @Test
+    public void testJwtWithRefreshTokensForOAuth2Enabled()
+            throws Exception
+    {
+        TestingHttpServer jwkServer = createTestingJwkServer();
+        jwkServer.start();
+        try (TokenServer tokenServer = new TokenServer(Optional.empty());
+                TestingTrinoServer server = TestingTrinoServer.builder()
+                        .setProperties(
+                                ImmutableMap.<String, String>builder()
+                                        .putAll(SECURE_PROPERTIES)
+                                        .put("http-server.authentication.type", "jwt,oauth2")
```

Comment:
Can test with the config that causes the issue i.e `http-server.authentication.type=oauth2,JWT` ?